### DISCUSSION TITLE
BT-15 (conditional) mandatory in subtyps 7-15

### DISCUSSION THREAD

**Discussion by JanEmde** [OP-TED/eForms-SDK#1046] [Q&A] [answered]
Field BT-15 is optional for subtypes 7-15 according to the Implementing Regulation. How is it that the field is (conditional) mandatory for the subtypes in the SDK?

**Comment by YvesJo** [comment, marked as answer]
Hi,
There is a business logic behind the conditional rules and this may be seen for example with form 16. BT-615 is relevant for restricted documents only while BT-15 is only relevant for non-restricted documents; also rules have been expressed based on their dependence on BT-14. 
The existence rule, as reported in the annex, is an aggregated expression that includes not only the rules inside the Business Group, but also the rule for the Business Group existence (e.g. optional x mandatory = optional). 
BT-14, which drives the existence of BT-15/BT-615, is optional for the 7-15 forms and already conveys the optional presence of the Business Group (i.e. existence of the Business Group = the one of BT-14). The context for BT-15 is the parent node which represents somehow the Business Group and the rule does not get fire if that node (i.e. that BG) does not exist.
KR

**Comment by JanEmde** [comment]
Thanks for the answer. Where exactly in the SDK is it defined that the business group is optional for the subtypes mentioned?

**Reply by YvesJo** [reply]
There is no such thing in the SDK as we do not manage BG and it is not always possible to have a direct parallel since the XML structure does not match 100% the annex tree.
Consideration of rules context is important to properly understand their reach. There are key elements in groups from which other elements depend, the existence of these last elements is then conditioned by the existence of the previous ones (cascade constraints involved by the business logic) and in eForms empty elements may not exist. 
These are all the elements to keep in mind when looking at rules.

**Reply by JanEmde** [reply]
That is really unfortunate. I thought the aim of the SDK was to enable the automatic generation of forms.
